chat-android/src/test/java/com/ably/chat/room/RoomFeatureSharedChannelTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.ably.chat.room
+
+import io.ably.lib.types.ChannelMode
+import io.ably.lib.types.ChannelOptions
+import io.mockk.every
+import io.mockk.verify
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert
+import org.junit.Test
+
+/**
+ * Test to check shared channel for room features.
+ * Spec: CHA-RC3
+ */
+class RoomFeatureSharedChannelTest {
+
+    @Test
+    fun `(CHA-RC3a) Features with shared channel should call channels#get only once with combined modes+options`() = runTest {
+        val mockRealtimeClient = createMockRealtimeClient()
+        val capturedChannelOptions = mutableListOf<ChannelOptions>()
+
+        every {
+            mockRealtimeClient.channels.get("1234::\$chat::\$chatMessages", any<ChannelOptions>())
+        } answers {
+            capturedChannelOptions.add(secondArg())
+            mockRealtimeClient.createMockChannel()
+        }
+
+        // Create room with all feature enabled,
+        val room = createMockRoom(realtimeClient = mockRealtimeClient)
+
+        // Messages, occupancy and presence features uses the same channel
+        Assert.assertEquals(room.messages.channel, room.presence.channel)
+        Assert.assertEquals(room.messages.channel, room.occupancy.channel)
+
+        // Reactions and typing uses independent channel
+        Assert.assertNotEquals(room.messages.channel, room.typing.channel)
+        Assert.assertNotEquals(room.messages.channel, room.reactions.channel)
+        Assert.assertNotEquals(room.reactions.channel, room.typing.channel)
+
+        Assert.assertEquals(1, capturedChannelOptions.size)
+        // Check for set presence modes
+        Assert.assertEquals(2, capturedChannelOptions[0].modes.size)
+        Assert.assertEquals(ChannelMode.presence, capturedChannelOptions[0].modes[0])
+        Assert.assertEquals(ChannelMode.presence_subscribe, capturedChannelOptions[0].modes[1])
```

Comment:
We don't want to use `Assert.assertTrue` because, it will not result in proper error message if one of the `mode`
is missing. So, it will be difficult to debug asertion error based on the error message.